### DISCUSSION TITLE
Multiple compute modules

### DISCUSSION THREAD

**Discussion by Bentschi** [TheGuyDanish/CM4_MATX#41] [Ideas]
Hello,

Just had the Idea that your ATX board could house multiple compute modules (I'm thinking about 2-4 compute modules).
A normal ATX power supply should provide more than enough power for multiple modules, and i think you would have the space to plan that in as an option.
Maybe make the other compute modules except the first one headless?

Anyway, i like the Idea of that project and I may steal some ideas for my PSOne RPi project if you are ok with that.

**Comment by derwana** [comment]
Housing multiple boards isn't our goal i think. The touringPi project already does this very well and is yet the way to go in this regard (clustering etc.).

**Comment by TheGuyDanish** [comment]
Hey Bentschi

Thanks for stopping by, and please, do use any part of CM4_MATX that might be suitable for your project. Sharing is caring!

As for the idea of a board with multiple compute modules, I did (at least a little bit) explore this with a coworker of mine, though with a more rackmount-oriented use-case in mind. A single PCB that hosts perhaps 4-5 CM4s, each with their own USB header for eMMC flashing and likely some connection for the NIC. That said, we haven't explored it a whole lot. Nor is it really in the cards for this project to expand in that direction.

Personally, I'd be happy to work on a board like that, but I would ideally like to get a finished revision of this board out the door first.

**Comment by Bentschi** [comment]
No, i didn't mean something crazy like the touringPi.
I really just thought about the bare minimum one could come up with to run another computer module headless next to the one that controls the board without extra IO and maybe only use the USB OTG function to flash the eMMC and then use the OTG as a network interface for that module.

**Reply by TheGuyDanish** [reply]
While there is some point to that, I don't think it would be used enough to warrant inclusion in the board, at least yet. Until we've got a first revision, I'm eager to keep costs down on the board, but this could be a niche thing that others could do with their own derivative boards.

**Reply by Bentschi** [reply]
Ok, you are right, if i think about it, what i just explained could be an "addon" for pretty much anything, one could make just a bare minimum 2.5" drive that houses a compute module and connects to the mainboard usb and uses the power from the atx psu.

**Reply by TheGuyDanish** [reply]
Exactly. I'm very amused by the idea of putting a CM4 on different formfactors. Maybe a 2.5" or 3.5" drive form factor.

Another idea I was toying with for a while was putting a CM4 on a PCI-e x1 card and use the slot-provided 12v to power the Pi through a voltage regulator. Then also adding a PCI->USB hub that could host a USB to Serial IC like the FTDI 2232 to communicate with the Pi from the host computer.

Not something I'll be working on for a while, I don't think, but the thought is quite interesting.

**Reply by derwana** [reply]
in regard of the 3.5" drive form factor: [mebs/CM4-NAS](https://github.com/mebs/CM4-NAS) already

> Removed the unnecessary IO and rearranged the remaining interfaces for a smaller footprint that would fit in the width of a standard 3.5" hard drive.

That project could be a great entrypoint to a 2.5"/3.5" project.
